numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1261,7 +1261,12 @@ arr_unravel_index(PyObject *self, PyObject *args, PyObject *kwds)
         goto fail;
     }
 
-    unravel_size = PyArray_MultiplyList(dimensions.ptr, dimensions.len);
+    unravel_size = PyArray_OverflowMultiplyList(dimensions.ptr, dimensions.len);
+    if (unravel_size == -1) {
+        PyErr_SetString(PyExc_ValueError,
+                        "dimensions greater than 'intp' is not supported");
```

Comment:
Really just a thought:
```suggestion
                        "dimensions are too large; arrays and shapes with 
                        "a total size "greater than 'intp' are not supported.");
```